eng/common/aotcompatibility/Check-AOT-Compatibility.ps1

```diff
@@ -0,0 +1,94 @@
+param([string]$ServiceDirectory, [string]$PackageName, [int]$NumAOTWarningsExpected=0)
+
+Write-Host "Creating a test app to publish."
+
+$csprojFile = "aotcompatibility.csproj"
+
+$csprojContent = @"
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net7.0</TargetFramework>
```

Comment:
Per Vitek, this should target net8.0 https://github.com/Azure/azure-sdk-for-net/pull/39428#issuecomment-1781625618
```suggestion
    <TargetFramework>net8.0</TargetFramework>
```

Reply:
I'm working on this in a separate branch because I ended up needing to change the approach a bit, but we can't target net 8 in the CI yet :(